src/transformers/integrations/flex_attention.py

```diff
@@ -0,0 +1,44 @@
+from typing import Optional, Tuple
+
+import torch
+
+from ..utils import is_torch_greater_or_equal
+
+
+if is_torch_greater_or_equal("2.5"):
```

Comment:
There is `is_torch_flex_attn_available()` in the utils. Might be more general and easier to change especially if conditions change on when flex attn is allowed.